src/networks/types.ts

```diff
@@ -0,0 +1,77 @@
+import { EthereumAddress } from '@/entities';
+import { GasPricesAPIData } from '@/entities/gas';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import { Chain } from '@wagmi/chains';
+// network.ts
+export enum Network {
+  arbitrum = 'arbitrum',
+  goerli = 'goerli',
+  mainnet = 'mainnet',
+  optimism = 'optimism',
+  polygon = 'polygon',
+  bsc = 'bsc',
+}
+
+export type NetworkTypes = 'layer1' | 'layer2' | 'testnet';
+
+export interface NetworkProperties extends Chain {
+  // network related data
+  enabled: boolean;
+  name: string;
+  longName: string;
+  value: Network;
+  networkType: 'layer1' | 'layer2' | 'testnet';
+  blockTimeInMs: number;
+
+  nativeCurrency: {
+    name: string;
+    symbol: string;
+    decimals: number;
+    address: string;
+    mainnetAddress?: string;
+  };
+
+  rpc: string;
+  getProvider: Promise<StaticJsonRpcProvider>;
+  balanceCheckerAddress: EthereumAddress;
+
+  // feature flags
+  features: {
+    txHistory: boolean;
+    flashbots: boolean;
+    walletconnect: boolean;
+    swaps: boolean;
+    nfts: boolean;
+    savings: boolean;
+    pools: boolean;
+    txs: boolean;
+  };
+
+  gas: {
+    speeds: string[];
+    gasType: 'eip1559' | 'legacy';
+
+    // for some networks gas is so cheap we dont want to round the gwei #
+    roundGasDisplay: boolean;
+
+    // this prob can just be blockTime
+    pollingIntervalInMs: number;
+
+    getGasPrices: () => Promise<GasPricesAPIData | null>;
+  };
+
+  swaps: {
+    defaultSlippage: number;
+    defaultToFastGas?: boolean;
+  };
+
+  nfts: {
+    simplehashNetwork?: string;
+  };
```

Comment:
im unsure how to handle, id love to use ethereum instead of mainnet and get rid of this change but afaik we have other instances were external deps or related use mainnet over ethereum


maybe we just add some logic to replace mainnet -> ethereum on the proxy?

Reply:
I have a path forward here but am not gonna change until after the zora release